test/expected/upsert.out

```diff
@@ -0,0 +1,108 @@
+INSERT INTO upsert_test_unique VALUES ('2017-01-20T09:00:01', 22.5, 'yellow') RETURNING *;
+           time           | temp | color  
+--------------------------+------+--------
+ Fri Jan 20 09:00:01 2017 | 22.5 | yellow
+(1 row)
+
+INSERT INTO upsert_test_unique VALUES ('2017-01-20T09:00:01', 21.2, 'brown');
+SELECT * FROM upsert_test_unique ORDER BY time, color DESC;
+           time           | temp | color  
+--------------------------+------+--------
+ Fri Jan 20 09:00:01 2017 | 22.5 | yellow
+ Fri Jan 20 09:00:01 2017 | 21.2 | brown
+(2 rows)
+
+INSERT INTO upsert_test_unique VALUES ('2017-01-20T09:00:01', 31.8, 'yellow') ON CONFLICT (time, color)
+DO UPDATE SET temp = 31.8;
+INSERT INTO upsert_test_unique VALUES ('2017-01-20T09:00:01', 54.3, 'yellow') ON CONFLICT DO NOTHING;
+SELECT * FROM upsert_test_unique ORDER BY time, color DESC;
+           time           | temp | color  
+--------------------------+------+--------
+ Fri Jan 20 09:00:01 2017 | 31.8 | yellow
+ Fri Jan 20 09:00:01 2017 | 21.2 | brown
+(2 rows)
+
+-- Test with multiple UNIQUE indexes
+CREATE TABLE upsert_test_multi_unique(time timestamp, temp float, color text);
+SELECT create_hypertable('upsert_test_multi_unique', 'time');
+ create_hypertable 
+-------------------
+ 
+(1 row)
+
+CREATE UNIQUE INDEX multi_time_temp_idx ON upsert_test_multi_unique (time, temp);
+CREATE UNIQUE INDEX multi_time_color_idx ON upsert_test_multi_unique (time, color);
+INSERT INTO upsert_test_multi_unique VALUES ('2017-01-20T09:00:01', 25.9, 'yellow');
+INSERT INTO upsert_test_multi_unique VALUES ('2017-01-21T09:00:01', 25.9, 'yellow');
+INSERT INTO upsert_test_multi_unique VALUES ('2017-01-20T09:00:01', 23.5, 'brown');
+INSERT INTO upsert_test_multi_unique VALUES ('2017-01-20T09:00:01', 25.9, 'purple') ON CONFLICT DO NOTHING;
+SELECT * FROM upsert_test_multi_unique ORDER BY time, color DESC;
+           time           | temp | color  
+--------------------------+------+--------
+ Fri Jan 20 09:00:01 2017 | 25.9 | yellow
+ Fri Jan 20 09:00:01 2017 | 23.5 | brown
+ Sat Jan 21 09:00:01 2017 | 25.9 | yellow
+(3 rows)
+
+INSERT INTO upsert_test_multi_unique VALUES ('2017-01-20T09:00:01', 25.9, 'blue') ON CONFLICT (time, temp)
+DO UPDATE SET color = 'blue';
+SELECT * FROM upsert_test_multi_unique ORDER BY time, color DESC;
+           time           | temp | color  
+--------------------------+------+--------
+ Fri Jan 20 09:00:01 2017 | 23.5 | brown
+ Fri Jan 20 09:00:01 2017 | 25.9 | blue
+ Sat Jan 21 09:00:01 2017 | 25.9 | yellow
+(3 rows)
+
+INSERT INTO upsert_test_multi_unique VALUES ('2017-01-21T09:00:01', 45.7, 'yellow') ON CONFLICT (time, color)
+DO UPDATE SET temp = 45.7;
+SELECT * FROM upsert_test_multi_unique ORDER BY time, color DESC;
+           time           | temp | color  
+--------------------------+------+--------
+ Fri Jan 20 09:00:01 2017 | 23.5 | brown
+ Fri Jan 20 09:00:01 2017 | 25.9 | blue
+ Sat Jan 21 09:00:01 2017 | 45.7 | yellow
+(3 rows)
+
```

Comment:
Might be interesting to see what happens if using the "wrong" on conflict clause. i.e. at the end `INSERT INTO upsert_test_multi_unique VALUES ('2017-01-20T09:00:01', 23.5, 'purple') ON CONFLICT (time, color) DO UPDATE set temp = 23.5` I would guess that would error out since that violates the (time,temp) unique constraint.